controller/lib/core/blower_fsm.h

```diff
@@ -159,19 +172,125 @@ class PressureControlFsm {
   Time start_time_;
   Time inspire_end_;
   Time expire_end_;
+  bool inspire_finished_ = false;
+  bool finished_ = false;
+};
+
+// "Breath finite state machine" for pressure assist mode.
+//
+// This mode is used for a patient that is not fully sedated and can initiate
+// breaths by himself, but once a breath has started, it is fully controlled
+// by the ventilator: inspiration time is set by the ventilator.
+// This mode enforce a minimum respiratory rate in case the patient does not
+// initiate enough breaths.
+// Similarly to pressure control, uses a square wave, but the duration of the
+// exhale leg can vary depending on the patient breath initiation, meaning the
+// breath can be finished (and the next one started) before the full breath
+// duration has passed.
+// In case the full breath duration passes, the fsm enforces a "mandatory
+// breath" to make sure the patient's RR does not go below the desired one.
+// Because the inhale is the first leg and only the exhale leg can vary in
+// duration, the first breath in this mode is always a "mandatory breath".
+class PressureAssistFsm {
+public:
+  explicit PressureAssistFsm(Time now, const VentParams &params)
+      : inspire_pressure_(cmH2O(static_cast<float>(params.pip_cm_h2o))),
+        expire_pressure_(cmH2O(static_cast<float>(params.peep_cm_h2o))),
+        start_time_(now), inspire_end_(start_time_ + InspireDuration(params)),
+        latest_expire_end_(inspire_end_ + MaxExpireDuration(params)) {}
```

Comment:
I'd suggest to rename this one to expire_deadline_ to clarify that it is a limit rather than a measurement.